src/allotropy/parsers/utils/pandas.py

```
@@ -279,7 +279,11 @@ def _key_matches(self, match_key: str, key: str) -> bool:
         # "++" can cause re.compile to fail. Since it is never a valid regex expression
         # it is safe to escape it to prevent the error.
         match_key = match_key.replace("++", r"\+\+")
-        return bool(re.fullmatch(match_key, key))
+        try:
+            return bool(re.fullmatch(match_key, key))
+        except re.error:
```

I'm concerned about silently skipping failing regexes, as this would mean a bad regex added to a parser would silently fail and just not pick up data.

I would prefer to diagnose the failing regex and fix it, as we do for ++ above.

That said, I checked out your branch and removed this to check the error myself, and found that there were no failures. 

I'm guessing that your other updates made it so we're no longer trying the bad regex? So I think OK to remove this for now.